docs/DATA_MODEL.md

```diff
@@ -0,0 +1,299 @@
+If none are available it will select any PropertyVariant with a valid Format.
+
+For example, given following PropertyVariants:
+
+* `description`
+* `description:lang=de:format=date`
+* `description:lang=en`
+* `description:lang=en:format=number`
+
+If the Publisher wants to show some property with the format `text` (aka no format Variant) and the user has selected
+the language `de`, the Publisher
+would first look at all PropertyVariants with `lang=de`. There is only one, `description:lang=de:format=date`, but it
+has the wrong format so the Publisher will try again with the default language, aka `lang=`.
+The only PropertyVariant without language is `description`, which has the correct format and is selected.
+
+For a second example, lets say the user still has selected the language `de` but the Publisher needs the format `number`
+to be able to show the property.
+In this case the first step is still the same, but the second step where we look at the PropertyVariant `description`
+now also fails because it has the wrong format.
+Now in the third step the Publisher will ignore the language format and will look at all PropertyVariants to find a
+valid one. In this
+example there is only one so it will select `description:lang=en:format=number`. If there would be more than one valid
+Variant with the same format priority, it will sort them alphabetically and select the first one. (Like
+the [KeyFilters](#keyfilters) specify.)
+
+## Variants and the iCal-Publisher
+
+Similar to the HTML-Publisher, only that it cannot render Markdown and will ignore this format.
+
+# Known Variants
+
+## Language Variant
+
+Has the VariantName "lang" and the value is a two-letter language abbreviation, like "en", "de", "fr", ...
+Denotes the language the value is in. Without a language variant it is assumed that this is the "default" language and
+will be used as a fallback if the wanted language is not available.
+
+## Format Variant
+
+Has the VariantName "format" and specifies what kind of data and in which format the Value is.
+
+Boudicca currently supports those formats:
+
+* Text (the default when no format is given, please do NOT use any VariantValue for this)
+* Numbers (VariantValue "number")
+* Dates (VariantValue "date")
+* Lists (VariantValue "list")
+* Markdown (VariantValue "markdown")
+
+### Text
+
+Format: Arbitrary text, no restrictions
+
+Sorting: Alphabetically, case-insensitive (so `aAbB` instead of `abAB`)
+
+Searching: Works with Equals and Contains queries.
+
+### Numbers
+
+Format: Allowed are integers, negative numbers and floating point numbers
+
+Examples:
+
+* 0
+* 1
+* -5
+* 0.5
+
+Sorting: By numerical value
+
+Searching: Works with Equals queries. Numbers need to be equal. TODO we probably need greater and smaller-queries here.
+
+### Dates
+
+Format: The format is the "Date and time with the offset" format of
+the [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) standard. TODO not sure if we really want to accept all ISO
```

Comment:
highlight light todo for possible upcoming "breaking" changes

Reply:
what exactly do you mean?

Reply:
in markdown, it should be nice if you add smth that highlights that there will changes via "!" emoji or bold text

Reply:
yeah, but i just made an decision and disallowed local dates :P